### DISCUSSION TITLE
The "dreaded" white screen - help

### DISCUSSION THREAD

**Discussion by ghost** [firefly-iii/firefly-iii#9181] [Firefly III]
Hey all.

I've self hosted some things before but usually simple stuff that don't need php. There are several moving parts and I'm not sure where the problem lies. I'm trying to run the bare metal binary on a local server (installed on Archlinux), however when I try to navigate to the url it's just a white screen.

DNS and port forwarding, and firewall are all fine. I'm using Caddy as the reverse proxy with the following config:

`finances.site.xyz {
root * /usr/share/webapps/firefly-iii/public
file_server
php_fastcgi unix//run/php-fpm/firefly.sock
}`

Most of the files/directories are owned by root user and root group, with the exception of **public** which I've given the Caddy group ownership on.

Php-fpm file/pool is set to listen for caddy:
user = caddy
group = caddy
listen.owner = caddy
listen.group = caddy
listen = /run/php-fpm/firefly.sock (which exists)

Playing around with it earlier I was able to get a message to show from Firefly with server 500 errors, but now I don't even get that. Please let me know what info you need from me. I used the steps outlined here: https://docs.firefly-iii.org/how-to/firefly-iii/installation/self-managed/

I'm probably missing something simple here like a permission issue, but not sure.

**Comment by JC5** [comment]
I expect that either the fpm logs or the nginx/caddy logs show the error.

**Reply by ghost** [reply]
Unfortunately no, just now I set Caddy to debug level of info and set php-fpm to write to a log file, restarted them, visited the site again ... no dice. The log files are surprisingly sparse. I tried on another device and on there I get error 500.

**Comment by ghost** [comment]
There is some things in the php-fpm log now.
FATAL:  role "root" does not exist

**Reply by JC5** [reply]
Mmmm, that's weird! Sorry, I'm not sure what this could be. Sorry :(
